### ISSUE TITLE
Can't pull container. failed to register layer: lsetxattr user.overlay.impure /etc: operation not supported

### ISSUE THREAD

**Issue by Gabgobie** [curl/curl-container#55]
Title pretty much says everything. As it seems this issue is with the container.

[They seem to have found a solution here](https://github.com/jsknnr/enshrouded-server/issues/56)

Other containers pull fine for me.

**Comment by dfandrich** [comment]
What command-line are you running that gives you this error?

**Comment by Gabgobie** [comment]
I'm using Bash

**Comment by dfandrich** [comment]
Please tell us what you typed in bash before seeing that error.

**Comment by Gabgobie** [comment]
docker pull <image>

```
root@docker-temp:~# docker pull curlimages/curl
Using default tag: latest
latest: Pulling from curlimages/curl
4abcf2066143: Already exists
1113023ab841: Extracting [==================================================>]  4.308MB/4.308MB
4ca545ee6d5d: Download complete
failed to register layer: lsetxattr user.overlay.impure /etc: operation not supported
root@docker-temp:~# docker pull quay.io/curl/curl:latest
latest: Pulling from curl/curl
b6f1a98f98c5: Already exists
1113023ab841: Extracting [==================================================>]  4.308MB/4.308MB
4ca545ee6d5d: Download complete
failed to register layer: lsetxattr user.overlay.impure /etc: operation not supported
root@docker-temp:~#
```

**Comment by dfandrich** [comment]
Are you running Docker inside a Docker container? I could see that causing
problems. In any case, this looks like a local configuration problem to me. It
works fine on my system:

$ docker pull curlimages/curl
Using default tag: latest
latest: Pulling from curlimages/curl
Digest: sha256:25d29daeb9b14b89e2fa8cc17c70e4b188bca1466086907c2d9a4b56b59d8e21
Status: Image is up to date for curlimages/curl:latest
docker.io/curlimages/curl:latest
$ docker pull quay.io/curl/curl:latest
latest: Pulling from curl/curl
Digest: sha256:1e136d717aab088c1f157ff816cf4219fe457a1687f11580e6a127488954f67f
Status: Downloaded newer image for quay.io/curl/curl:latest
quay.io/curl/curl:latest

**Comment by Gabgobie** [comment]
https://docs.docker.com/engine/release-notes/25.0/#extended-attributes-for-tar-files

Do you use tar archives that were created on MacOS? The docker release notes seem to blame the MacOS tar implementation

**Comment by xquery** [comment]
cannot repro ... you might also try running podman

**Comment by Furisto** [comment]
I see the same issue. The latest release that works is 8.1.1. Every release that I have tested after that has this problem.

**Comment by xquery** [comment]
are you (@Furisto ) also on ubuntu ?

**Comment by aquarta** [comment]
I see the same issue, I'm using a brand new workspace in gitpod
Linux fiware-tutorialsiotsens-r1iaycrgx85 6.1.75-060175-generic #202402010024 SMP PREEMPT_DYNAMIC Thu Feb  1 01:44:23 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.4 LTS"

**Comment by xquery** [comment]
so ubuntu it is then, will take a deeper look

**Comment by aquarta** [comment]
Thanks, I forgot to report the docker version

docker --version
Docker version 26.0.1, build d260a54

The pull is working fine with 7.88.1 version

**Comment by xquery** [comment]
This issue between docker (some version) and ubuntu (some version) ... we have not made any changes in terms of the container itself and more about how buildah does things ... the problem is that latest docker no longer ignores a certain class of errors and we need to rebuild using a fixed version of buildah.
 
In the short term, one can rollback to some older version of docker or use podman ... will refresh the buildah toolchain and arrange a new release which will hopefully address.

**Comment by Gabgobie** [comment]
Thanks a lot for taking the time to look into it. Feel free to ping me so I can tell you if it's fixed when you're ready

**Comment by xquery** [comment]
does the latest release fix things for you ?

**Comment by chadlwilson** [comment]
Is there any reason to believe this would be a problem only with the arm64 image or when running cross-arch via QEMU (e.g arm64 image on amd64)?

I have issues with the curl image inside a multi-stage Dockerfile - but only cross-arch and only since [buildkit 0.17.0](https://github.com/moby/buildkit/releases/tag/v0.17.0) released a few days ago (i.e `0.16.0` is OK)

Running within a dind environment with rootless buildkit builder, host kernel `Linux 5.10.219-208.866.amzn2.x86_64 amd64`.

**Comment by chadlwilson** [comment]
In any case, there seems to be still be a problem here when using the curl image alongside rootless dind to build images (e.g ones using the curl image in a multi-stage build file) in particular when using `fuse-overlayfs` with buildx/buildkit.

It seems right now we'd need the curl image to be built with buildah `1.35+` to avoid this problem as it appears no fix has been backported to older buildah versions. Currently this project is using `1.23.1` off Ubuntu Jammy repositories. Even Ubuntu 24.04 Noble doesn't have a fixed buildah I think, as the issue hasn't been backported to the `1.33.x` line.

Other than building an arbitrary buildah version from source to use to then build curl images, I've tried seeing if a backport can be done at https://github.com/containers/storage/issues/2166 (which could then be pulled into buildah `1.33.x` with a patch update of the storage libraries)

**Comment by xquery** [comment]
any chance to check now with 8.12.0 ?

**Comment by chadlwilson** [comment]
I don't use this image in a dind environment now, but I suspect still an issue as don't think the fix has been backported to buildah 1.33.x for use in `ubuntu-latest`/`24.04`

**Comment by xquery** [comment]
thx for the info!

**Comment by xquery** [comment]
thanks to everyone pitching in onthis - going to close this bug and raise a new bug about installing custom build of buildah

**Comment by philippebi** [comment]
It seems the issue is still not resolved? If I set the `nouser_xattr` flag on the `/var/lib/docker` FS, I hit the same issue with the latest curl image (8.13 at the moment):

```
root@ubuntu2204:~# mount | grep /var/lib/docker
/dev/mapper/dockervg-dockerlv on /var/lib/docker type ext4 (rw,relatime,nouser_xattr)
root@ubuntu2204:~# docker pull curlimages/curl
Using default tag: latest
latest: Pulling from curlimages/curl
f18232174bc9: Pull complete 
0840d8ec766c: Extracting [==================================================>]   5.91MB/5.91MB
bd9ddc54bea9: Download complete 
failed to register layer: lsetxattr /etc: xattr "user.overlay.impure": operation not supported
```

**Comment by philippebi** [comment]
Yes, because you probably used an underlying filesystem that supports user extended attributes, such as the default ext4. However, if you mount `/var/lib/docker` on tmpfs, it likely won’t work anymore on Ubuntu 22.04.

**Comment by xquery** [comment]
closing this as latest ubuntu has much fresher version of buildah which will hopefully fix this
